src/main/java/org/folio/search/utils/CallNumberUtils.java

```diff
@@ -178,19 +174,13 @@ public static List<CallNumberBrowseItem> excludeIrrelevantResultItems(String cal
       return records;
     }
 
-    records.forEach(r -> {
-      r.getInstance().setItems(getItemsFiltered(callNumberType, r));
-      r.getInstance().setHoldings(getHoldingsFiltered(callNumberType, r));
-    });
-    return records;
-  }
-
-  @NotNull
-  private static List<@Valid Holding> getHoldingsFiltered(String callNumberType, CallNumberBrowseItem item) {
-    return item.getInstance().getHoldings()
+    records.forEach(r -> r.getInstance().setItems(getItemsFiltered(callNumberType, r)));
+    return records
       .stream()
-      .filter(h -> isInGivenType(callNumberType, h.getCallNumber()))
-      .toList();
+      .filter(r -> r.getInstance().getItems()
+        .stream()
+        .anyMatch(i -> r.getFullCallNumber() == null
+          || i.getEffectiveCallNumberComponents().getCallNumber().equals(r.getFullCallNumber()))).toList();
```

Comment:
This check is added in order to exclude non-matching call numbers, which an resulting instance, with matching call number, may include.